waspc/data/Generator/templates/server/src/routes/auth/passport/google/google.js

```diff
@@ -0,0 +1,76 @@
+import express from 'express'
+import passport from 'passport'
+import GoogleStrategy from 'passport-google-oauth20'
+
+import waspServerConfig from '../../../../config.js'
+import { contextWithUserEntity, authConfig, findOrCreateUserByExternalAuthAssociation } from '../../utils.js'
+import { sign } from '../../../../core/auth.js'
+import { configFn, getUserFields } from './googleConfig.js'
+
+const config = ((config) => {
+  if (!config?.clientId) {
+    throw new Error("auth.google.configFn must return an object with a clientId property.")
+  }
+
+  if (!config?.clientSecret) {
+    throw new Error("auth.google.configFn must return an object with a clientSecret property.")
+  }
+
+  if (!config?.scope) {
+    throw new Error("auth.google.configFn must return an object with a scope property.")
+  } else if (!Array.isArray(config.scope) || !config.scope.includes('profile')) {
+    throw new Error("auth.google.configFn returned an object with an invalid scope property. It must be an array including 'profile'.")
+  }
+
+  return config
+})(configFn())
+
+passport.use('waspGoogleLoginStrategy', new GoogleStrategy({
+  clientID: config.clientId,
+  clientSecret: config.clientSecret,
+  callbackURL: `${waspServerConfig.frontendUrl}/auth/login/google`,
+  scope: config.scope,
+  passReqToCallback: true
+}, addGoogleProfileToRequest))
+
+// This function is invoked after we successfully exchange the one-time-use OAuth code for a real Google API token.
+// This token was used to get the Google profile information supplied as a parameter.
+// We add the Google profile to the request for downstream use.
+async function addGoogleProfileToRequest(req, _accessToken, _refreshToken, googleProfile, done) {
+  req.wasp = { ...req.wasp, googleProfile }
```

Comment:
I tried to really simplify this function to just give us back the google profile (by adding it to the request). This makes it easier to follow the downstream usage and flow I think.

Reply:
It is just the shape of the callback for Google's Passport implementation. The library invokes this.